### PR TITLE
ucx/verbs: added support for interrupt moderation

### PR DESCRIPTION
**Pull request by hoopoepg**
- added support for interrupt moderation according to user config
- updated configuration scripts to detect CQ_MODERATION
  capabilities

fix for https://github.com/openucx/ucx/issues/1769

### CODE REVIEW

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -81,6 +81,16 @@ ucs_config_field_t uct_ib_iface_config_table[] = {
    "Number of send WQEs for which completion is requested.",
    ucs_offsetof(uct_ib_iface_config_t, tx.cq_moderation), UCS_CONFIG_TYPE_UINT},
 
+#ifdef HAVE_DECL_IBV_EXP_CQ_MODERATION
```
Comment (hoopoepg) [inline src/uct/ib/base/ib_iface.c]: not sure if we should hide config parameters on non-support EXP_MODERATION

Comment (brminich) [reply src/uct/ib/base/ib_iface.c]: I'd hide, like it is done now

File: src/uct/ib/base/ib_iface.c [hunk 2]
```diff
@@ -81,6 +81,16 @@ ucs_config_field_t uct_ib_iface_config_table[] = {
...
+  {"TX_CQ_MODERATION_COUNT", "-1",
+   "Number of send WQEs for which completion is requested.",
+   ucs_offsetof(uct_ib_iface_config_t, tx.cq_moderation_count), UCS_CONFIG_TYPE_INT},
+
+  {"TX_CQ_MODERATION_PERIOD", "-1",
+   "Number of send WQEs for which completion is requested.",
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: Wrong desc

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4408/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c (hunk 2 above)
Comment (hoopoepg) [reply src/uct/ib/base/ib_iface.c]: description is updated

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4419/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c [hunk 3]
```diff
@@ -560,6 +570,20 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
...
     self->config.max_inl_resp = inl;
 
+#if HAVE_DECL_IBV_EXP_CQ_MODERATION
+    if (config->tx.cq_moderation_count >= 0 || config->tx.cq_moderation_period >= 0) {
+        struct ibv_exp_cq_attr cq_attr = {
+            .comp_mask = IBV_EXP_CQ_ATTR_MODERATION,
+            .moderation.cq_count = (uint16_t)ucs_max(config->tx.cq_moderation_count, 0),
+            .moderation.cq_period = (uint16_t)ucs_max(config->tx.cq_moderation_period, 0)
+        };
+        if (ibv_exp_modify_cq(self->send_cq, &cq_attr, IBV_EXP_CQ_MODERATION)) {
```
Comment (miked-mellanox) [inline src/uct/ib/base/ib_iface.c]: plz add ucs_error with error code

Comment (hoopoepg) [reply src/uct/ib/base/ib_iface.c]: done

**Comment by brminich** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4422/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4426/ for details (Mellanox internal link).

**Pull request merged**